### PR TITLE
Implemented login page

### PR DESCRIPTION
**Pull request by qnewald**
Implemented and styled login page.

### CODE REVIEW

File: client/src/ts/members/Login.tsx [hunk 1]
```diff
@@ -1,20 +1,72 @@
...
+    },
+  }));
+
+  const classes = useStyles();
+  
+  
   const [username, setUsername] = useState<string>("");
   const [password, setPassword] = useState<string>("");
+  const [capslock, setCapslock] = useState<boolean>(false);
+  const [failedLogin, setFailedLogin] = useState<boolean>(false);
```
Comment (christophtrybek) [inline client/src/ts/members/Login.tsx:69]: * Failed Login message is not displayed correctly when incorrect password is submitted

File: client/src/ts/members/Login.tsx [hunk 2]
```diff
@@ -1,20 +1,72 @@
...
+      pointerEvents: 'none',
+    },
+  }));
+
+  const classes = useStyles();
+  
+  
   const [username, setUsername] = useState<string>("");
   const [password, setPassword] = useState<string>("");
+  const [capslock, setCapslock] = useState<boolean>(false);
```
[the remark this answers is not in the document]
Comment (qnewald) [reply client/src/ts/members/Login.tsx:68]: JS can only check for Caps Lock on key events

File: client/src/ts/members/Login.tsx [hunk 3]
```diff
@@ -4,60 +4,155 @@
...
+      <Textfield className={classes.inputfield} id="password" label="Passwort" type="password" value={password}
+        onChange = {event => {setPassword(event.target.value);}}
+        onKeyUp={event => {handleKeyUp(event.getModifierState("CapsLock"));}} fullWidth />
+    );
+  };
+
+  /**
+   * Check if Capslock is enabled
+   */
+  const handleKeyUp = (isCapsEnabled:boolean) => {
```
Comment (LennartRukower) [inline client/src/ts/members/Login.tsx:111]: Consider changing it to sth. like:
```
const handleKeyUp = (event: React.KeyboardEvent<HTMLInputElement>) => {
  setCapslock(event.getModifierState("CapsLock"));
}
```

File: client/src/ts/members/Login.tsx [hunk 4]
```diff
@@ -4,60 +4,155 @@
...
+  /*
+   * Gets the password correct state of the password field, depending on if a previous login attempt failed
+   */
+  const getPasswordField: VoidFunction = () => {
+    if (failedLogin) {
+      return (
+        <Textfield error className={classes.inputfield} id="password" label="Passwort" type="password"
+          helperText="Passwort oder Benutzername sind nicht korrekt" value={password}
+          onChange = {event => {setPassword(event.target.value);}}
+          onKeyUp={event => {handleKeyUp(event.getModifierState("CapsLock"));}} fullWidth />
```
Comment (LennartRukower) [inline client/src/ts/members/Login.tsx:98]: Consider changing it to: 
`onKeyUp={handleKeyUp} `

File: client/src/ts/members/Login.tsx [hunk 5]
```diff
@@ -4,60 +4,155 @@
...
+      );
+    }
+    return (
+      <Textfield className={classes.inputfield} id="password" label="Passwort" type="password" value={password}
+        onChange = {event => {setPassword(event.target.value);}}
+        onKeyUp={event => {handleKeyUp(event.getModifierState("CapsLock"));}} fullWidth />
```
Comment (LennartRukower) [inline client/src/ts/members/Login.tsx:104]: Consider changing it to: 
`onKeyUp={handleKeyUp} `

File: client/src/ts/members/Login.tsx [hunk 6]
```diff
@@ -4,60 +4,155 @@
...
-      </Card>
+      <Grid container spacing={0} alignItems="center" justify="center">
+        <Grid item xs={10} sm={8} md={6} lg={4}>
+          <Paper className={classes.paper}>
+            <img className={classes.logo} src={logo}/>
+            <h1>Login</h1>
+            <form className={classes.login} onSubmit={event => {login();}}>
+              <Textfield className={classes.inputfield} id="username" label="Benutzername" type="text" value={username}
+                onChange = {event => {setUsername(event.target.value);}}
+                onKeyUp={event => {handleKeyUp(event.getModifierState("CapsLock"));}} fullWidth />
```
Comment (LennartRukower) [inline client/src/ts/members/Login.tsx:135]: Consider changing it to: 
`onKeyUp={handleKeyUp} `

**Comment by LennartRukower** [comment]
You should also link the corresponding issue(s).
https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue

File: client/src/ts/members/Login.tsx [hunk 7]
```diff
@@ -5,65 +5,160 @@ import React, { useState, useContext } from "react";
 import { useHistory } from "react-router-dom";
 
 import api from "../utils/api";
-import {AuthContext} from "../global/AuthContext";
```
Comment (LennartRukower) [inline client/src/ts/members/Login.tsx:8]: `import { AuthContext } from "../global/AuthContext";` is missing.

**Comment by LennartRukower** [comment]
Changes were made to "develop". Please merge "develop" into this branch. 

Tipp: Use Sourcetree from Atlassian (_https://www.sourcetreeapp.com/_)

Here is how you merge a branch into your own in Sourcetree:
_https://confluence.atlassian.com/get-started-with-sourcetree/merge-changes-from-one-branch-to-another-git-847359121.html_

File: client/src/ts/members/Login.tsx [hunk 8]
```diff
@@ -5,65 +5,160 @@ import React, { useState, useContext } from "react";
...
         setAuthenticated(true);
         history.push("/");
       } else {
-        console.log("Login Failed");
+        setFailedLogin(true);
       }
-      console.log(res.data);
-    }, (err) => {
-      console.log(err);
+    }, () => {
```
Comment (christophtrybek) [inline client/src/ts/members/Login.tsx:89]: Please use explicit .catch call

File: client/src/ts/members/Login.tsx [hunk 9]
```diff
@@ -5,65 +5,160 @@ import React, { useState, useContext } from "react";
...
+  };
+
+  /**
+   * Check if Capslock is enabled
+   */
+  const handleKeyUp = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    setCapslock(event.getModifierState("CapsLock"));
+  };
+
+
```
Comment (christophtrybek) [inline client/src/ts/members/Login.tsx:121]: Remove new line

**Pull request merged**